### PR TITLE
Merge Destroy- and CreateResource warheads

### PR DESCRIPTION
**Pull request by reaperrr**
Not only does this mean one less warhead to maintain, it also now allows to reliably remove pre-existing resources before spawning new ones, without having to fiddle with delays or order of warheads to ensure the freshly added resources aren't immediately removed again.

### CODE REVIEW

File: OpenRA.Mods.Common/UpdateRules/Rules/MergeResourceWarheads.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			var destroyResourceWarheads = weaponNode.ChildrenMatching("Warhead").Where(wh => wh.Value.Value == "DestroyResource");
+			foreach (var drwh in destroyResourceWarheads)
+			{
+				drwh.ReplaceValue("ModifyResource");
+				drwh.AddNode("RemoveResource", "true");
+			}
+
+			var createResourceWarheads = weaponNode.ChildrenMatching("Warhead").Where(wh => wh.Value.Value == "CreateResource");
+			foreach (var crwh in createResourceWarheads)
+				crwh.ReplaceValue("ModifyResource");
```
Comment (atlimit8) [inline OpenRA.Mods.Common/UpdateRules/Rules/MergeResourceWarheads.cs:42]: It is `RemoveResources` in `ModifyResourceWarhead`.
```suggestion
				crwh.ReplaceValue("ModifyResources");
```

File: mods/cnc/weapons/superweapons.yaml [hunk 2]
```diff
@@ -30,9 +30,10 @@ Atomic:
 			Light: 60
 			Heavy: 50
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@4Res_areanukea: DestroyResource
+	Warhead@4Res_areanukea: ModifyResource
 		Size: 3
 		Delay: 3
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/cnc/weapons/superweapons.yaml:36]:
```suggestion
		RemoveResources: true
```

File: mods/cnc/weapons/superweapons.yaml [hunk 3]
```diff
@@ -55,9 +56,10 @@ Atomic:
 			Light: 60
 			Heavy: 50
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@8Res_areanukeb: DestroyResource
+	Warhead@8Res_areanukeb: ModifyResource
 		Size: 4
 		Delay: 6
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/cnc/weapons/superweapons.yaml:62]:
```suggestion
		RemoveResources: true
```

File: mods/cnc/weapons/superweapons.yaml [hunk 4]
```diff
@@ -76,9 +78,10 @@ Atomic:
 			Light: 60
 			Heavy: 50
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@11Res_areanukec: DestroyResource
+	Warhead@11Res_areanukec: ModifyResource
 		Size: 5
 		Delay: 9
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/cnc/weapons/superweapons.yaml:84]:
```suggestion
		RemoveResources: true
```

File: mods/cnc/weapons/superweapons.yaml [hunk 5]
```diff
@@ -106,9 +109,10 @@ IonCannon:
 		InvalidTargets: Vehicle, Structure, Wall, Husk, Trees, Creep
 		Size: 1
 		Delay: 3
-	Warhead@4Res_area2: DestroyResource
+	Warhead@4Res_area2: ModifyResource
 		Size: 2
 		Delay: 6
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/cnc/weapons/superweapons.yaml:115]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/explosions.yaml [hunk 6]
```diff
@@ -204,7 +204,8 @@ CrateNuke:
 			Concrete: 25
 		AffectsParent: true
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@2Res_impact: DestroyResource
+	Warhead@2Res_impact: ModifyResource
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/explosions.yaml:208]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/explosions.yaml [hunk 7]
```diff
@@ -219,9 +220,10 @@ CrateNuke:
 			Concrete: 25
 		AffectsParent: true
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@5Res_areanuke1: DestroyResource
+	Warhead@5Res_areanuke1: ModifyResource
 		Size: 4
 		Delay: 5
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/explosions.yaml:226]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/explosions.yaml [hunk 8]
```diff
@@ -251,8 +253,9 @@ MiniNuke:
 			Concrete: 25
 		AffectsParent: true
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@2Res_impact: DestroyResource
+	Warhead@2Res_impact: ModifyResource
 		Size: 1
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/explosions.yaml:258]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/explosions.yaml [hunk 9]
```diff
@@ -268,9 +271,10 @@ MiniNuke:
 			Concrete: 25
 		AffectsParent: true
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@5Res_areanuke1: DestroyResource
+	Warhead@5Res_areanuke1: ModifyResource
 		Size: 2
 		Delay: 5
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/explosions.yaml:277]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/explosions.yaml [hunk 10]
```diff
@@ -293,9 +297,10 @@ MiniNuke:
 		Delay: 10
 		ValidTargets: Trees
 		DamageTypes: Incendiary
-	Warhead@9Res_areanuke2: DestroyResource
+	Warhead@9Res_areanuke2: ModifyResource
 		Size: 3
 		Delay: 10
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/explosions.yaml:303]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/explosions.yaml [hunk 11]
```diff
@@ -313,9 +318,10 @@ MiniNuke:
 		Delay: 15
 		ValidTargets: Trees
 		DamageTypes: Incendiary
-	Warhead@12Res_areanuke3: DestroyResource
+	Warhead@12Res_areanuke3: ModifyResource
 		Size: 4
 		Delay: 15
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/explosions.yaml:324]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/superweapons.yaml [hunk 12]
```diff
@@ -41,8 +41,9 @@ Atomic:
 			Wood: 25
 			Concrete: 25
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@2Res_impact: DestroyResource
+	Warhead@2Res_impact: ModifyResource
 		Size: 1
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/superweapons.yaml:46]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/superweapons.yaml [hunk 13]
```diff
@@ -62,9 +63,10 @@ Atomic:
 			Wood: 25
 			Concrete: 25
 		DamageTypes: Prone50Percent, TriggerProne, FireDeath, Incendiary
-	Warhead@6Res_areanuke1: DestroyResource
+	Warhead@6Res_areanuke1: ModifyResource
 		Size: 2
 		Delay: 5
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/superweapons.yaml:69]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/superweapons.yaml [hunk 14]
```diff
@@ -91,9 +93,10 @@ Atomic:
 		Delay: 15
 		ValidTargets: Trees
 		DamageTypes: Incendiary
-	Warhead@11Res_areanuke2: DestroyResource
+	Warhead@11Res_areanuke2: ModifyResource
 		Size: 3
 		Delay: 10
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/superweapons.yaml:99]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/superweapons.yaml [hunk 15]
```diff
@@ -115,9 +118,10 @@ Atomic:
 		Delay: 15
 		ValidTargets: Trees
 		DamageTypes: Incendiary
-	Warhead@15Res_areanuke3: DestroyResource
+	Warhead@15Res_areanuke3: ModifyResource
 		Size: 4
 		Delay: 15
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/superweapons.yaml:124]:
```suggestion
		RemoveResources: true
```

File: mods/ra/weapons/superweapons.yaml [hunk 16]
```diff
@@ -139,9 +143,10 @@ Atomic:
 		Delay: 20
 		ValidTargets: Trees
 		DamageTypes: Incendiary
-	Warhead@19Res_areanuke4: DestroyResource
+	Warhead@19Res_areanuke4: ModifyResource
 		Size: 5
 		Delay: 20
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ra/weapons/superweapons.yaml:149]:
```suggestion
		RemoveResources: true
```

File: mods/ts/weapons/otherweapons.yaml [hunk 17]
```diff
@@ -60,8 +60,9 @@ Bomb:
 	Warhead@4Smu: LeaveSmudge
 		SmudgeType: MediumCrater
 		InvalidTargets: Vehicle, Building, Wall
-	Warhead@5Res: DestroyResource
+	Warhead@5Res: ModifyResource
 		Size: 1
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ts/weapons/otherweapons.yaml:65]:
```suggestion
		RemoveResources: true
```

File: mods/ts/weapons/superweapons.yaml [hunk 18]
```diff
@@ -26,7 +26,8 @@ MultiCluster:
 		ImpactActors: false
 		Explosions: large_twlt
 		ImpactSounds: expnew09.aud
-	Warhead@ResourceDestruction: DestroyResource
+	Warhead@ResourceDestruction: ModifyResource
+		RemoveResource: true
```
Comment (atlimit8) [inline mods/ts/weapons/superweapons.yaml:30]:
```suggestion
		RemoveResources: true
```

**Review by atlimit8 — CHANGES_REQUESTED** [review]
It looks like you forgot the `s` in the rule and then applied it.

**Comment by reaperrr** [comment]
Leaving the singular vs plural issue aside, I'm beginning to wonder whether it would be smarter to tackle the TODOs while I'm at it, otherwise we'd have to replace the boolean again in a future refactor. I have other stuff in the pipeline that's higher priority for me though, so this will probably stay stale for a while.

**Comment by reaperrr** [comment]
Will probably take some time until I get back to this, closing for now.

**Pull request closed without merging**